src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs

```diff
@@ -79,6 +79,32 @@ public void EmitDebugInfoForSourceTextWithoutEncoding2()
 </symbols>", options: PdbValidationOptions.ExcludeMethods);
         }
 
+        [Fact]
+        public void EmitDebugInfoForSynthesizedSyntaxTree()
+        {
+            var tree1 = SyntaxFactory.ParseCompilationUnit(@"
+#line 1 ""test.cs""
+class C { void M() {} }
+").SyntaxTree;
+            var tree2 = SyntaxFactory.ParseCompilationUnit(@"
+class D { void M() {} }
+").SyntaxTree;
+
+            var comp = CSharpCompilation.Create("test", new[] { tree1, tree2 }, TargetFrameworkUtil.StandardReferences, TestOptions.DebugDll);
+
+            var result = comp.Emit(new MemoryStream(), pdbStream: new MemoryStream());
+            result.Diagnostics.Verify();
+
+            comp.VerifyPdb(@"
```

Comment:
Nit: Consider using raw strings to avoid the need for escaping nested quotes.

Reply:
I don't think we should use raw strings for these tests unless we change all the tests. The test helpers print out escaped quotes automatically. They would need to be updated to not do so.